diff --git a/dist/vite.js b/dist/vite.js
index 67b5167d2da535d2e1166546b331d52762f81f28..d50203cc73cb4fdef400c683cb8723129db50696 100644
--- a/dist/vite.js
+++ b/dist/vite.js
@@ -17,7 +17,7 @@ function node(options) {
 				async handler(id, importer) {
 					let importerResolvedId;
 					if (options?.importer) importerResolvedId = (await this.resolve(options.importer))?.id;
-					const resolved = await this.resolve("@universal-deploy/node/serve", importerResolvedId ?? importer);
+					const resolved = (importerResolvedId ?? importer) && await this.resolve(importerResolvedId ?? importer);
 					if (!resolved) try {
 						return { id: createRequire(import.meta.url).resolve("@universal-deploy/node/serve") };
 					} catch {
