diff --git a/dist/index.js b/dist/index.js
index 14f80545c97fc416de930971f4a017f97f09deb7..63693ca45da0787920eb96c8b9e2ce73f787fd26 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -110,7 +110,35 @@ async function findSupportedDeploymentTargets(thisNodePlugin, c) {
 	if (plugins.some((p) => p.name.startsWith("ud:node:emit") && p[INSTANCE] !== thisNodePlugin?.[INSTANCE])) found.push("@universal-deploy/node");
 	return found;
 }
-
+function target(entry) {
+	let resolvedEntry
+	return {
+		name: "ud:target:emit",
+		apply: "build",
+		config: {
+			order: "post",
+			handler() {
+				return { environments: { ssr: { build: { rolldownOptions: { input: { index: entry } } } } } }
+			}
+		},
+		async buildStart() {
+      const resolved = await this.resolve(entry);
+      if (resolved) {
+        resolvedEntry = resolved.id;
+      }
+    },
+		transform(code, id) {
+			if (resolvedEntry && id === resolvedEntry) {
+				const s = new MagicString(code)
+				s.prepend(`import "virtual:ud:catch-all";\n`)
+				return {
+					code: s.toString(),
+					map: s.generateMap({ hires: true })
+				}
+			}
+		}
+	}
+}
 //#endregion
 //#region src/plugins/auto.ts
 /**
@@ -121,11 +149,7 @@ function auto(options) {
 	return [
 		catchAll(),
 		devServer(),
-		...node$1(options?.node).map((p) => {
-			p[INSTANCE] = instance;
-			return enablePluginIf((config) => noDeploymentTargetFound(p, config), p);
-		}),
-		...netlifyGlue()
+		target(options.entry),
 	];
 }
 
