diff --git a/dist/node/vite/plugins/pluginUniversalDeploy/getServerConfig.js b/dist/node/vite/plugins/pluginUniversalDeploy/getServerConfig.js
index d415ea9465996a28b294f0e4ba3a9cc4929bd26e..eb0ce77c7961935217bb73f58352d6336f124b76 100644
--- a/dist/node/vite/plugins/pluginUniversalDeploy/getServerConfig.js
+++ b/dist/node/vite/plugins/pluginUniversalDeploy/getServerConfig.js
@@ -1,3 +1,4 @@
+import { dirname, resolve } from 'node:path';
 export { getServerConfig };
 export { isUniversalDeployVitePreview };
 import { catchAllEntry } from '@universal-deploy/store';
@@ -7,6 +8,7 @@ function getServerConfig(vikeConfig) {
     let serverEntryId;
     let serverFilePath = null;
     let serverEntryVike;
+    let isServerEntry = false;
     // universal-deploy support must be manually enabled
     const serverConfig = 
     // +config.js > `export default { server: true }`
@@ -21,8 +23,10 @@ function getServerConfig(vikeConfig) {
         assert('filePathAbsoluteFilesystem' in serverPlusFile.definedAt);
         serverFilePath = serverPlusFile.definedAt.filePathAbsoluteFilesystem;
         assert(serverFilePath);
-        serverEntryId = serverFilePath;
-        serverEntryVike = serverFilePath;
+        const entry = serverPlusFile.valueIsLoaded && serverPlusFile.value && typeof serverPlusFile.value === 'object' ? serverPlusFile.value.entry : undefined;
+        isServerEntry = typeof entry === 'string';
+        assert(entry === undefined || isServerEntry);
+        serverEntryVike = serverEntryId = isServerEntry ? resolve(dirname(entry)) : serverFilePath;
     }
     else {
         serverEntryId = catchAllEntry;
@@ -36,6 +40,7 @@ function getServerConfig(vikeConfig) {
         // It either points to the default fetchable endpoint (vike/fetch), or one defined by the user through +server.
         serverEntryVike,
         serverFilePath,
+        isServerEntry,
     };
 }
 function isUniversalDeployVitePreview(vikeConfig, viteConfigResolved) {
diff --git a/dist/node/vite/plugins/pluginUniversalDeploy.js b/dist/node/vite/plugins/pluginUniversalDeploy.js
index 2519e3934414ccf8915fd4c49c7be230a42d0478..2c54325964fb597d84d7261b4d4ed6a678aba0fa 100644
--- a/dist/node/vite/plugins/pluginUniversalDeploy.js
+++ b/dist/node/vite/plugins/pluginUniversalDeploy.js
@@ -24,7 +24,7 @@ function pluginUniversalDeploy(vikeConfig) {
                 assertUsage(target === undefined, `${target} requires +server — see https://vike.dev/server`);
             }),
         ];
-    const { serverEntryVike, serverEntryId, serverFilePath } = serverConfig;
+    const { serverEntryVike, serverEntryId, serverFilePath, isServerEntry } = serverConfig;
     return [
         ...universalDeploy(),
         {
