diff --git a/dist/node/vite/plugins/pluginUniversalDeploy.js b/dist/node/vite/plugins/pluginUniversalDeploy.js
index 2519e3934414ccf8915fd4c49c7be230a42d0478..57adbf302ae69456520c9e7fc9d9dc3733800b57 100644
--- a/dist/node/vite/plugins/pluginUniversalDeploy.js
+++ b/dist/node/vite/plugins/pluginUniversalDeploy.js
@@ -26,7 +26,7 @@ function pluginUniversalDeploy(vikeConfig) {
         ];
     const { serverEntryVike, serverEntryId, serverFilePath } = serverConfig;
     return [
-        ...universalDeploy(),
+        ...pluginGetServerEntry(),
         {
             name: 'vike:pluginUniversalDeploy:entries',
             config() {
@@ -57,3 +57,32 @@ function pluginUniversalDeploy(vikeConfig) {
         !serverFilePath ? null : pluginUnwrapProdOptions(serverFilePath),
     ].filter((p) => p !== null);
 }
+
+function pluginGetServerEntry(vikeConfig) {
+  if (vikeConfig.buildServerIndex === false) return []
+  if (vikeConfig.serverEntrypoint) return [
+    {
+      name: 'vike:emit-custom-server-entrypoint',
+      apply: 'build',
+      config() {
+        return {
+          environments: {
+            ssr: {
+              resolve: {
+                noExternal: true
+              },
+              build: {
+                rolldownOptions: {
+                  input: {
+                    index: vikeConfig.serverEntrypoint
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  ]
+  return universalDeploy()
+}
