diff --git a/dist/node/vite/plugins/pluginUniversalDeploy.js b/dist/node/vite/plugins/pluginUniversalDeploy.js
index 2519e3934414ccf8915fd4c49c7be230a42d0478..3271a4340192c4b2b9a5e25da64e8417e2f27c4b 100644
--- a/dist/node/vite/plugins/pluginUniversalDeploy.js
+++ b/dist/node/vite/plugins/pluginUniversalDeploy.js
@@ -1,7 +1,7 @@
 export { pluginUniversalDeploy };
 import { toRou3 } from 'convert-route';
 import { addEntry } from '@universal-deploy/store';
-import universalDeploy, { resolveTargets } from '@universal-deploy/vite';
+import { resolveTargets } from '@universal-deploy/vite';
 import { pluginServerEntryInject } from './pluginUniversalDeploy/pluginServerEntryInject.js';
 import { getDeployConfig } from './pluginUniversalDeploy/getDeployConfig.js';
 import { pluginCommon } from './pluginUniversalDeploy/common.js';
@@ -26,7 +26,6 @@ function pluginUniversalDeploy(vikeConfig) {
         ];
     const { serverEntryVike, serverEntryId, serverFilePath } = serverConfig;
     return [
-        ...universalDeploy(),
         {
             name: 'vike:pluginUniversalDeploy:entries',
             config() {
