--- a/dist/node/vite/plugins/pluginUniversalDeploy.js
+++ b/dist/node/vite/plugins/pluginUniversalDeploy.js
@@ -27,6 +27,16 @@ function pluginUniversalDeploy(vikeConfig) {
     const { serverEntryVike, serverEntryId, serverFilePath } = serverConfig;
     return [
         ...universalDeploy(),
+        {
+            name: 'ud:target:emit',
+            apply: 'build',
+            config: {
+                order: 'post',
+                handler() {
+                    return { environments: { ssr: { build: { rolldownOptions: { input: { index: './src/server/entrypoint.ts' } } } } } }
+                }
+            }
+        },
         {
             name: 'vike:pluginUniversalDeploy:entries',
             config() {
